### PR TITLE
[SYCLomatic] Report warning of --query-api-mapping option conflict.

### PR DESCRIPTION
**Pull request by tangjj11**
Signed-off-by: Tang, Jiajun jiajun.tang@intel.com

### CODE REVIEW

File: clang/lib/DPCT/DPCT.cpp [hunk 1]
```diff
@@ -253,6 +253,18 @@ UnifiedPath getCudaInstallPath(int argc, const char **argv) {
   return Path;
 }
 
+bool hasOptConflictWithQuery(int argc, const char **argv) {
```
Comment (zhimingwang36) [inline clang/lib/DPCT/DPCT.cpp:258]:
```suggestion
// Please add comment for the functionality
bool checkOptionConflictForActionQuery(int argc, const char **argv) {
```

File: clang/lib/DPCT/DPCT.cpp [hunk 2]
```diff
@@ -843,6 +855,17 @@ int runDPCT(int argc, const char **argv) {
 
   std::vector<std::string> SourcePathList;
   if (QueryAPIMapping.getNumOccurrences()) {
+    if (QueryAPIMapping.getNumOccurrences() > 1) {
+      llvm::outs() << "Warning: Each query focuses on one API. Only the last "
```
Comment (zhimingwang36) [inline clang/lib/DPCT/DPCT.cpp]:
```suggestion
      llvm::outs() << "Warning: Option --query-api-mapping is specified multi time, only the last one is used, all other are ignored. "
```

File: clang/lib/DPCT/DPCT.cpp [hunk 3]
```diff
@@ -843,6 +855,17 @@ int runDPCT(int argc, const char **argv) {
...
+                      "one will be queried.\n";
+    }
+    if (hasOptConflictWithQuery(argc, argv)) {
+      llvm::outs()
+          << "Warning: When using --query-api-mapping, only "
```
Comment (zhimingwang36) [inline clang/lib/DPCT/DPCT.cpp]: "Warning: For API mapping query function triggered by option --query-api-mapping, all other option are ignored except "--extra-arg" and "--cuda-include-path".

Comment (tangjj11) [reply clang/lib/DPCT/DPCT.cpp]: Now the logic is just exit here but not ignore, if we need ignore, we need clear some option variable here be cause the `CommonOptionsParser::create` has done.

File: clang/lib/DPCT/DPCT.cpp (hunk 1 above)
Comment (tangjj11) [reply clang/lib/DPCT/DPCT.cpp:258]: fixed

File: clang/lib/DPCT/DPCT.cpp (hunk 2 above)
Comment (tangjj11) [reply clang/lib/DPCT/DPCT.cpp]: fixed

File: clang/lib/DPCT/DPCT.cpp [hunk 4]
```diff
@@ -843,6 +857,19 @@ int runDPCT(int argc, const char **argv) {
 
   std::vector<std::string> SourcePathList;
   if (QueryAPIMapping.getNumOccurrences()) {
+    if (QueryAPIMapping.getNumOccurrences() > 1) {
+      llvm::outs()
+          << "Warning: Option --query-api-mapping is specified multi time, "
```
Comment (ShengchenJ) [inline clang/lib/DPCT/DPCT.cpp]:
```suggestion
          << "Warning: Option --query-api-mapping is specified multi times, "
```

Comment (tangjj11) [reply clang/lib/DPCT/DPCT.cpp]: fixed

**Review by ShengchenJ — APPROVED** [review]

File: clang/lib/DPCT/DPCT.cpp [hunk 5]
```diff
@@ -843,6 +857,19 @@ int runDPCT(int argc, const char **argv) {
...
+      llvm::outs()
+          << "Warning: Option --query-api-mapping is specified multi times, "
+             "only the last one is used, all other are ignored.\n";
+    }
+    if (hasOptConflictWithQuery(argc, argv)) {
+      llvm::outs()
+          << "Warning: For API mapping query function triggered by option "
+             "--query-api-mapping, only \"--extra-arg\" and "
+             "\"--cuda-include-path\" option can be set.\n";
+      ShowStatus(MigrationErrorConflictOptions);
```
Comment (zhimingwang36) [inline clang/lib/DPCT/DPCT.cpp:871]: For API mapping query, only option --extra-arg and --cuda-include-path can be used together with option --query-api-mapping.

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**